thirdparty/catch2/build.gradle

```diff
@@ -0,0 +1,58 @@
+apply plugin: 'cpp'
+apply plugin: 'visual-studio'
+apply plugin: 'edu.wpi.first.NativeUtils'
+
+ext {
+    nativeName = 'catch2'
+}
+
+apply from: "${rootDir}/shared/config.gradle"
+
+def catch2UserConfigInput = file("src/main/native/include/catch2/catch_user_config.hpp.in")
+def catch2UserConfigOutput  = file("$buildDir/generated-includes/catch2/catch_user_config.hpp")
+
+task generateCatchUserConfig() {
+    description = 'Generates the Catch2 user config file.'
+    group = 'catch2'
+
+    outputs.file catch2UserConfigOutput
+    inputs.file catch2UserConfigInput
+
+    doLast {
+        println "Writing config to $catch2UserConfigOutput"
+
+        if (catch2UserConfigOutput.exists()) {
+            catch2UserConfigOutput.delete()
+        }
+
+        def read = catch2UserConfigInput.text
+        read = read.replace('${CATCH_CONFIG_DEFAULT_REPORTER}', "console")
+        read = read.replace('${CATCH_CONFIG_CONSOLE_WIDTH}', "80")
+        read = read.replaceAll("#cmakedefine", "#define")
```

Comment:
There are options that are mutually exclusive with each other, so a simple find-replace won't work. You'll need to manually patch the header and select the options we want (0 by default is probably a good starting point).